src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefConditionalOperatorTests.cs

```diff
@@ -1015,6 +1015,241 @@ .maxstack  1
+                        System.Console.WriteLine(y); //expected 5 - but we get 2
+                    }
+                }
+                """;
+
+            var comp = CompileAndVerify(source,
+                expectedOutput: """
+                5
+                1
+                5
+                """);
+            comp.VerifyDiagnostics();
+
+            comp.VerifyIL("C.Main", @"
+{
+  // Code size      122 (0x7a)
+  .maxstack  4
+  .locals init (string V_0, //str
+                int V_1, //x
+                int V_2, //y
+                int V_3, //z
+                int& V_4,
+                int V_5,
+                bool V_6)
+  IL_0000:  ldstr      ""a2""
+  IL_0005:  stloc.0
+  IL_0006:  ldc.i4.1
+  IL_0007:  stloc.1
+  IL_0008:  ldc.i4.2
+  IL_0009:  stloc.2
+  IL_000a:  ldc.i4     0x309
+  IL_000f:  stloc.3
+  IL_0010:  ldloc.0
+  IL_0011:  ldstr      ""whatever""
+  IL_0016:  call       ""bool string.op_Equality(string, string)""
+  IL_001b:  brfalse.s  IL_0023
+  IL_001d:  ldloca.s   V_1
+  IL_001f:  stloc.s    V_4
+  IL_0021:  br.s       IL_0053
+  IL_0023:  ldloc.0
+  IL_0024:  brfalse.s  IL_0044
+  IL_0026:  ldloc.0
+  IL_0027:  callvirt   ""int string.Length.get""
+  IL_002c:  stloc.s    V_5
+  IL_002e:  ldloc.s    V_5
+  IL_0030:  ldc.i4.2
+  IL_0031:  blt.s      IL_0044
+  IL_0033:  ldloc.s    V_5
+  IL_0035:  ldc.i4.s   10
+  IL_0037:  ble.s      IL_003f
+  IL_0039:  ldloc.s    V_5
+  IL_003b:  ldc.i4.s   22
+  IL_003d:  bne.un.s   IL_0044
+  IL_003f:  ldc.i4.1
+  IL_0040:  stloc.s    V_6
+  IL_0042:  br.s       IL_0047
+  IL_0044:  ldc.i4.0
+  IL_0045:  stloc.s    V_6
+  IL_0047:  ldloc.s    V_6
+  IL_0049:  brtrue.s   IL_004f
+  IL_004b:  ldloca.s   V_3
+  IL_004d:  br.s       IL_0051
+  IL_004f:  ldloca.s   V_2
+  IL_0051:  stloc.s    V_4
+  IL_0053:  ldloc.s    V_4
+  IL_0055:  dup
+  IL_0056:  dup
+  IL_0057:  ldind.i4
+  IL_0058:  ldc.i4.1
+  IL_0059:  add
+  IL_005a:  stind.i4
+  IL_005b:  dup
+  IL_005c:  dup
+  IL_005d:  ldind.i4
+  IL_005e:  ldc.i4.1
+  IL_005f:  add
+  IL_0060:  stind.i4
+  IL_0061:  dup
+  IL_0062:  dup
+  IL_0063:  ldind.i4
+  IL_0064:  ldc.i4.1
+  IL_0065:  add
+  IL_0066:  stind.i4
+  IL_0067:  ldind.i4
+  IL_0068:  call       ""void System.Console.WriteLine(int)""
+  IL_006d:  ldloc.1
+  IL_006e:  call       ""void System.Console.WriteLine(int)""
+  IL_0073:  ldloc.2
+  IL_0074:  call       ""void System.Console.WriteLine(int)""
+  IL_0079:  ret
+}
+");
+        }
+
+        [Fact]
+        public void TestReadonlyRefConditionalWithSpilling()
+        {
+            var source = """
+                string str = "a2";
+                int x = 1;
+                int y = 2;
+
+                ref readonly int roy = ref x;
+
+                ref readonly int r =
+                      ref str is "whatever" ? ref roy
+                    : ref str is { Length: >= 2 and <= 10 or 22 } ? ref y
+                    : ref System.Runtime.CompilerServices.Unsafe.NullRef<int>();
```

Comment:
It looks like this test is failing in CI because ```Unsafe``` type isn't available in ```TargetFramework.NetLatest``` when the test is executed on Desktop framework. Consider using ```TargetFramework.NetCoreApp``` instead and conditionally supplying expected output:
```
                expectedOutput: !ExecutionConditionUtil.IsMonoOrCoreClr ? null : """
                5
                1
                5
                """);
```